opentelemetry-api/src/opentelemetry/baggage/__init__.py

```diff
@@ -81,16 +81,7 @@ def set_baggage(
         A Context with the value updated
     """
     baggage = dict(get_all(context=context))
-    if not _is_valid_key(name):
-        _logger.warning(
-            "Baggage key `%s` does not match format, ignoring", name
-        )
-    elif not _is_valid_value(str(value)):
-        _logger.warning(
-            "Baggage value `%s` does not match format, ignoring", value
-        )
-    else:
-        baggage[name] = value
+    baggage[name] = value
```

Comment:
We are validating pairs only during `extract` now correct?

Reply:
correct

Reply:
This is a public API and there should be validation here regardless of the prior validation in propagator?

Reply:
@srikanthccv I added [this](https://github.com/open-telemetry/opentelemetry-python/pull/3058/files#diff-e1157703561645c8bebb82b3d93b1566cf34e9b83f2abb13ae268f64548b4e6aR250) test case. Should we be able to call `set_baggage` with a string that has spaces to create a context and then use that context to call the propagator `inject`? If so, then we need to remove the check in `set_baggage`.

Reply:
The value given in `set_baggage` must match the regex. It should be a URL encoded by the user if needed (in this case, where you have spaces). It can be decoded while added to the baggage list and encoded when serialised.

Reply:
I believe this comment is still relevant and needs to be addressed/discussed.

Reply:
@srikanthccv please read [this](https://github.com/open-telemetry/opentelemetry-python/issues/2934#issuecomment-1319385776).

Reply:
I re-read the specification (the baggage format applies to header value but anything can be added to the bagge). I looked at the reference implementation tests here https://github.com/w3c/baggage/tree/main/test/baggage. This is correct.